### PR TITLE
ARROW-11632: [Rust] Make csv::Reader propagate schema metadata to generated RecordBatches

### PR DESCRIPTION
**Pull request by mkmik**

### CODE REVIEW

**Review by alamb — COMMENTED** [review]
Thanks @mkmik  -- The code looks great

Can you please add a test for this behavior (mostly so that someone doesn't break it inadvertently in the future)? It looks like this module has a bunch of examples that you can probably repurpose.

https://github.com/apache/arrow/blob/356c300c5ee1e2b23a83652514af11e3a731d596/rust/arrow/src/csv/reader.rs#L841

**Review by jorgecarleitao — APPROVED** [review]
LGTM. Thanks!

**Comment by mkmik** [comment]
test added. PTAL

**Comment by mkmik** [comment]
@alamb friendly ping

**Comment by jorgecarleitao** [comment]
I only now realized that this is your first contribution, @mkmik ! Thank you very much for your interest on the arrow and for your PR! 👍 

We usually give it some time before merging so that other contributors have the opportunity to take a look and comment.

**Comment by mkmik** [comment]
:-) no rush; just thought it might have got lost in the pipes

File: rust/arrow/src/csv/reader.rs [hunk 1]
```diff
@@ -838,6 +843,38 @@ mod tests {
...
+    fn test_csv_schema_metadata() {
+        let mut metadata = std::collections::HashMap::new();
+        metadata.insert("foo".to_owned(), "bar".to_owned());
+        let schema = Schema::new_with_metadata(
+            vec![
+                Field::new("city", DataType::Utf8, false),
+                Field::new("lat", DataType::Float64, false),
+                Field::new("lng", DataType::Float64, false),
+            ],
+            metadata.clone(),
```
Comment (alamb) [inline rust/arrow/src/csv/reader.rs:856]: 👍

**Review by alamb — APPROVED** [review]
Sorry for the delay @mkmik  -- I am settling into a routine where I try and clear up Arrow related stuff in the afternoon/evening my timezone. 

I think this PR is simple enough that it doesn't any more discussion at this time. If anyone spots an issue, let us know and we can fix it in a follow on PR.

**Comment by alamb** [comment]
Thanks @mkmik  -- great first contribution

**Pull request closed without merging**